javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -662,23 +662,40 @@ public T get(int index) {
     }
 
     @Override
-    public Array<T> dropUntil(Predicate<? super T> predicate) {
+    public Array<T> dropWhile(Predicate<? super T> predicate) {
         Objects.requireNonNull(predicate, "predicate is null");
-        return dropWhile(predicate.negate());
+        return dropUntil(predicate.negate());
     }
 
     @Override
-    public Array<T> dropWhile(Predicate<? super T> predicate) {
+    public Array<T> dropUntil(Predicate<? super T> predicate) {
         Objects.requireNonNull(predicate, "predicate is null");
         for (int i = 0; i < length(); i++) {
-            if (!predicate.test(get(i))) {
+            if (predicate.test(get(i))) {
                 return drop(i);
             }
         }
         return empty();
     }
 
     @Override
+    public Array<T> dropRightWhile(Predicate<? super T> predicate) {
+        Objects.requireNonNull(predicate, "predicate is null");
+        return dropRightUntil(predicate.negate());
+    }
+
+    @Override
+    public Array<T> dropRightUntil(Predicate<? super T> predicate) {
+        Objects.requireNonNull(predicate, "predicate is null");
+        for (int i = length() - 1; i >= 0; i--) {
```

Comment:
yes, that would be great!

we can remove all the Objects.requireNonNull checks of drop*Until and drop*While then. It is sufficient to have the checks within Collections:

```java
    @SuppressWarnings("unchecked")
    static <T, S extends IndexedSeq<T>> S dropUntil(S empty, S seq, Predicate<? super T> predicate) {
        Objects.requireNonNull(predicate, "predicate is null");
        for (int i = 0; i < seq.length(); i++) {
            if (predicate.test(seq.get(i))) {
                return (S) seq.drop(i);
            }
        }
        return empty;
    }
```